.github/workflows/ci.yaml

```diff
@@ -16,6 +16,7 @@ jobs:
         ruby-version:
         - '2.7'
         - '3.0'
+        - '3.1'
```

Comment:
You can drop 2.7 here too, since that might save some cycles

Reply:
👍 This is a temporary measure so I can avoid regressions in 2.7 while working on this PR.  I don't intend to keep 3 rubies.

Reply:
When we delete Ruby 2.7 we can also drop the gems in #22761 I think

Reply:
no. the current 3.1 versions out there do not solve those problems and are still needed. I commented in that PR to hopefully answer that question